### PR TITLE
Add `gnmic diff set-to-notifs` subcommand

### LINKED ISSUE #80: `gnmic diff` for comparing SetRequests, SetRequest & Notifications
I'm wondering if gnmic would be open to incorporating this functionality that's currently implemented in ygot here (README has examples): https://github.com/openconfig/ygot/tree/master/gnmidiff

Proposed commands (open to changing command names):

### Compares intent between two SetRequests
`gnmic diff setrequest cmd/demo/setrequest2.textproto cmd/demo/setrequest.textproto`

### Compares whether after a SetRequest has been applied, the notifications that you get back from the device are consistent with the input SetRequest
`gnmic diff setrequest-to-notifs cmd/demo/setrequest.textproto cmd/demo/notifs.textproto`

Any suggestions on how to improve usability are appreciated. I think the meaning of these commands might be fuzzy at first, especially the last one. I'm thinking of just making it clear in the command description.

### PR DESCRIPTION
**Pull request by wenovus**
```
$ gnmic diff set-to-notifs --setrequest cmd/demo/setrequest.textproto --response cmd/demo/subscriberesponses.textproto

SetToNotifsDiff(-want/SetRequest, +got/Notifications):
- /lacp/interfaces/interface[name=Port-Channel9]/config/interval: "FAST"
- /lacp/interfaces/interface[name=Port-Channel9]/config/name: "Port-Channel9"
- /lacp/interfaces/interface[name=Port-Channel9]/name: "Port-Channel9"
- /network-instances/network-instance[name=VrfBlue]/config/name: "VrfBlue"
- /network-instances/network-instance[name=VrfBlue]/config/type: "openconfig-network-instance-types:L3VRF"
- /network-instances/network-instance[name=VrfBlue]/name: "VrfBlue"
m /system/config/hostname:
  - "violetsareblue"
  + "rosesarered"

$ gnmic diff set-to-notifs --setrequest cmd/demo/setrequest.textproto --response cmd/demo/getresponse.textproto

SetToNotifsDiff(-want/SetRequest, +got/Notifications):
- /lacp/interfaces/interface[name=Port-Channel9]/config/interval: "FAST"
- /lacp/interfaces/interface[name=Port-Channel9]/config/name: "Port-Channel9"
- /lacp/interfaces/interface[name=Port-Channel9]/name: "Port-Channel9"
- /network-instances/network-instance[name=VrfBlue]/config/name: "VrfBlue"
- /network-instances/network-instance[name=VrfBlue]/config/type: "openconfig-network-instance-types:L3VRF"
- /network-instances/network-instance[name=VrfBlue]/name: "VrfBlue"
m /system/config/hostname:
  - "violetsareblue"
  + "rosesarered"
```

Fixes #80

### CODE REVIEW

File: cmd/demo/diff-subcommands.md
Comment (karimra) [inline cmd/demo/diff-subcommands.md:1]: Thanks for writing docs as well, the documentation files should be under `/docs` and referenced in https://github.com/openconfig/gnmic/blob/main/mkdocs.yml. 
For the commands reference it should be somewhere around here: https://github.com/openconfig/gnmic/blob/main/mkdocs.yml#L120

Comment (wenovus) [reply cmd/demo/diff-subcommands.md:1]: Thanks for the pointers, added. I split the diff documentation into parts like the generate command. Not sure how to test the website though, is there a commandline script I can call?

Comment (karimra) [reply cmd/demo/diff-subcommands.md:1]: I usually run this `docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material:8.3.4` from the repo root. The website is reachable on `localhost:8000`

Comment (wenovus) [reply cmd/demo/diff-subcommands.md:1]: Thanks! tested.

**Review by karimra — APPROVED** [review]

**Pull request merged**